### ISSUE TITLE
Defend against padding instability

### ISSUE THREAD

**Issue by jswrenn** [google/zerocopy#1648]
In light of https://github.com/rust-lang/rust/pull/129778, we should do more to defend against misconceptions about the durability of padding bytes across moves. For example, `FromZeros::new_zeroed` returns its value by-move, and so padding bytes within `Self` may not be `0`.

We can do some combination of the following. First, we can aggressively sign-post this risk. As a rule, one *shouldn't* depend on the value of padding bytes. 

Second, we can wrap zeroed types in a type that ensures padding remains preserved; e.g.:
```rust
union PaddingStable<T> {
    a: core::mem::ManuallyDrop<T>,
    b: [u8; core::mem::size_of::<u8>()]
}
```

**Comment by RalfJung** [comment]
```rust
union PaddingStable<T> {
    a: core::mem::ManuallyDrop<T>,
    b: [u8; core::mem::size_of::<u8>()]
}
```
The last `u8` should be `T`, right?

**Comment by jswrenn** [comment]
Yeah, that's a typo. Unfortunately, using `T` there also isn't currently stable. For now, the best we can do is documentation, I think (https://github.com/google/zerocopy/pull/1660).

**Comment by joshlf** [comment]
If we need to add a utility type, we can do that when the need arises.

For now, I'll consider this closed by #1660.
